Kernel/Storage/ATA/AHCI/Controller.cpp

```diff
@@ -167,12 +167,13 @@ UNMAP_AFTER_INIT ErrorOr<void> AHCIController::initialize_hba(PCI::DeviceIdentif
     u32 version = hba().control_regs.version;
 
     hba().control_regs.ghc = 0x80000000; // Ensure that HBA knows we are AHCI aware.
-    PCI::enable_interrupt_line(device_identifier());
     PCI::enable_bus_mastering(device_identifier());
+    TRY(reserve_irqs(1, true));
+    auto irq = TRY(allocate_irq(0));
```

Comment:
This `TRY` can probably be a `MUST` if `reserve_irqs` succeeded earlier?

Reply:
Yeah, that makes sense. I will change that!